test/e2e_node/runtime_conformance_test.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	retryTimeout = time.Minute * 4
+	pollInterval = time.Second * 5
+)
+
+type testStatus struct {
+	Name          string
+	RestartPolicy api.RestartPolicy
+	Phase         api.PodPhase
+	State         ContainerState
+	RestartCount  int
+	Ready         bool
+}
+
+var _ = Describe("Container runtime Conformance Test", func() {
+	var cl *client.Client
+
+	BeforeEach(func() {
+		// Setup the apiserver client
+		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
+	})
+
+	Describe("container runtime conformance blackbox test", func() {
+		Context("when start a container that exits success", func() {
+			var terminateContainers []ConformanceContainer
+			It("it should run with the expected status [Conformance]", func() {
+				testContainer := api.Container{
+					Image:           "gcr.io/google_containers/busybox",
+					Command:         []string{"sh", "-c", "env"},
+					ImagePullPolicy: api.PullIfNotPresent,
+				}
+				testStatuss := []testStatus{
+					{"busybox-terminate-always", api.RestartPolicyAlways, api.PodRunning, ContainerStateWaiting | ContainerStateRunning | ContainerStateTerminated, -1, false},
+					{"busybox-terminate-onfailure", api.RestartPolicyOnFailure, api.PodSucceeded, ContainerStateTerminated, 0, false},
+					{"busybox-terminate-never", api.RestartPolicyNever, api.PodSucceeded, ContainerStateTerminated, 0, false},
+				}
+				for _, testStatus := range testStatuss {
+					testContainer.Name = testStatus.Name
+					terminateContainer := ConformanceContainer{testContainer, cl, testStatus.RestartPolicy, *nodeName}
+					terminateContainers = append(terminateContainers, terminateContainer)
+					err := terminateContainer.Create()
+					Expect(err).NotTo(HaveOccurred())
+					time.Sleep(time.Second * 60)
```

Comment:
If each time we create container, we need to sleep 60s, the test will be quite slow.